src/main/java/com/atlassian/bitbucket/jenkins/internal/util/GitSCMUtil.java

```diff
@@ -0,0 +1,18 @@
+package com.atlassian.bitbucket.jenkins.internal.util;
+
+import hudson.plugins.git.GitSCM;
+import org.eclipse.jgit.transport.RemoteConfig;
+
+public class GitSCMUtil {
+
+    private GitSCMUtil() { }
+
+    /**
+     * Extracts a remote config representation of the BitbucketRepository from a GitSCM
+     * @param gitSCM a GitSCM attached to a BitbucketSCM
+     * @return the SCM representation of the repository
+     */
+    public static RemoteConfig getGitRepository(GitSCM gitSCM) {
+        return gitSCM.getRepositories().get(0);
```

Comment:
I think this is a bit dangerous, given there is a chance we will support multiple repos per SCM in the future. There is no need for this at all, we can iterate over the Collection with a `forEach()` or some such thing, and if the branch name starts with the repo name, remove it from the beginning (and break out of the for loop).

Reply:
That's fair enough. I copied the pattern we used in the `BitbucketSCM.getRepository` where we do the same thing. This is a little different though, in the GitSCM not the BitbucketSCM, so happy to replace with a for loop (I like keeping in a utility method though, as we may need to update it in the future and it seems out of place in the PostBuildStatus class)

Reply:
Interesting. Yeah, we seem to have this pattern in a few places and as Daniel said, using a Collection `repositories` in the serialized class was to save us data migration headaches in the future.
It's up to you if you wanna keep this or try iterating over the Collection.
Either way, I feel like a util class for getting the first item in a collection is a _touch_ overkill, especially given we don't seem to be using this util class for anything else.
But this sort of thing is a bit subjective and there's no hard science for when a util class is justified - it's in the same category as "when is a hill a mountain?", if you know what I mean ;-)
Personally, my (subjective) criteria for creating a util/helper class is (just off the top of my head):
- Does it remove a fair bit of clutter that would be in the inline code?
- More importantly, does it remove _repeated_ clutter from the inline code?
- etc.

Reply:
It fails by both of those metrics- I think my motivation was more about encapsulation. But I don't feel strongly enough about it to press the point.